### ISSUE TITLE
"catch_exception" not functioning with v3 API

### ISSUE THREAD

**Issue by KentonParton** [fivetran/great_expectations#3205]
**Describe the bug**
I am using the v3 API, SparkDFExecutionEngine with the following classes
ExpectationConfiguration
RuntimeBatchRequest
context.get_validator().validate(). 

When I include "catch_exception" in the expectation kwargs or in context.get_validator().validate(catch_exceptions=True), an exception is still thrown.

```
Calculating Metrics:  50%|█████     | 4/8 [00:00<00:00,  5.30it/s]Traceback (most recent call last):
  File "/opt/project/src/main/glue/reproduce.py", line 92, in <module>
    expectation_suite=suite,
  File "/usr/local/lib/python3.6/site-packages/great_expectations/validator/validator.py", line 1209, in validate
    "result_format": result_format,
  File "/usr/local/lib/python3.6/site-packages/great_expectations/validator/validator.py", line 474, in graph_validate
    metrics = self.resolve_validation_graph(graph, metrics, runtime_configuration)
  File "/usr/local/lib/python3.6/site-packages/great_expectations/validator/validator.py", line 519, in resolve_validation_graph
    runtime_configuration=runtime_configuration,
  File "/usr/local/lib/python3.6/site-packages/great_expectations/validator/validator.py", line 560, in _resolve_metrics
    metrics_to_resolve, metrics, runtime_configuration
  File "/usr/local/lib/python3.6/site-packages/great_expectations/execution_engine/execution_engine.py", line 282, in resolve_metrics
    **metric_provider_kwargs
  File "/usr/local/lib/python3.6/site-packages/great_expectations/expectations/metrics/metric_provider.py", line 58, in inner_func
    return metric_fn(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/great_expectations/expectations/metrics/map_metric_provider.py", line 465, in inner_func
    message=f'Error: The column "{column_name}" in BatchData does not exist.'
great_expectations.exceptions.exceptions.ExecutionEngineError: Error: The column "unknown_column" in BatchData does not exist.
Calculating Metrics:  50%|█████     | 4/8 [00:00<00:00,  4.45it/s]
```

**To Reproduce**
Code to reproduce the behavior:
```python
from great_expectations.data_context import BaseDataContext
from great_expectations.data_context.types.base import InMemoryStoreBackendDefaults
from great_expectations.data_context.types.base import DataContextConfig
from great_expectations.core import ExpectationSuite, ExpectationConfiguration
from great_expectations.core.batch import RuntimeBatchRequest
from pyspark.sql import SparkSession
import pandas as pd

data_context_config = DataContextConfig(
    datasources={
        "spark_datasource": {
            "execution_engine": {
                "class_name": "SparkDFExecutionEngine",
                "module_name": "great_expectations.execution_engine",
            },
            "class_name": "Datasource",
            "module_name": "great_expectations.datasource",
            "data_connectors": {
                "runtime_data_connector": {
                    "class_name": "RuntimeDataConnector",
                    "batch_identifiers": [
                        "domain_id",
                        "component_name"
                    ]
                }
            }
        }
    },
    validation_operators={
        "action_list_operator": {
            "class_name": "ActionListValidationOperator",
            "action_list": [
                {
                    "name": "store_validation_result",
                    "action": {"class_name": "StoreValidationResultAction"},
                },
                {
                    "name": "store_evaluation_params",
                    "action": {"class_name": "StoreEvaluationParametersAction"},
                },
                {
                    "name": "update_data_docs",
                    "action": {"class_name": "UpdateDataDocsAction"},
                },
            ],
        }
    },
    expectations_store_name="expectations_store",
    validations_store_name="validations_store",
    evaluation_parameter_store_name="evaluation_parameter_store",
    checkpoint_store_name="checkpoint_store",
    store_backend_defaults=InMemoryStoreBackendDefaults(),
)

context = BaseDataContext(project_config=data_context_config)
suite: ExpectationSuite = context.create_expectation_suite("suite", overwrite_existing=True)

expectation_configuration = ExpectationConfiguration(
    expectation_type='expect_column_values_to_not_be_null',
    kwargs={
        'catch_exceptions': True,  # expect exceptions to be caught
        'result_format': 'SUMMARY',
        'include_config': False,
        'column': 'unknown_column'  # intentionally incorrect column to force error
    },
    meta={
        "Notes": "Some notes"
    }
)
suite.add_expectation(expectation_configuration=expectation_configuration)

pandasDF = pd.DataFrame(data=[['Scott'], ['Jeff'], ['Thomas'], ['Ann']], columns=['Name'])

spark = SparkSession.builder.appName("local").getOrCreate()
sparkDF = spark.createDataFrame(pandasDF)
sparkDF.show()

runtime_batch_request = RuntimeBatchRequest(
    datasource_name="spark_datasource",
    data_connector_name="runtime_data_connector",
    data_asset_name="insert_your_data_asset_name_here",
    runtime_parameters={
        "batch_data": sparkDF
    },
    batch_identifiers={
        "domain_id": "ininfsgi283",
        "component_name": "some_component",
    }
)
validator = context.get_validator(
    batch_request=runtime_batch_request,
    expectation_suite=suite,
).validate()
results = validator.results
print(results)

```

**Environment (please complete the following information):**
 - Operating System: MacOS
 - Great Expectations Version: 0.13.26

Thanks for your time.

**Comment by talagluck** [comment]
Thanks for submitting, @KentonParton ! We will review internally and respond over the next week or so.

**Comment by talagluck** [comment]
Hi @KentonParton! Following up, this represents a gap between the V2 Batch Kwargs and V3 Batch Request APIs. This feature has not yet been implemented in V3, though this has been added to the prioritization queue for our next sprint planning. It may take several weeks for us to get there, and this goes pretty deep in the code, but it is being prioritized in the near future, and someone will follow up when this is resolved. In the meantime, please don't hesitate to reach out if you have any additional questions. Thanks for your patience!

**Comment by KentonParton** [comment]
Thanks for letting me know @talagluck. 

We are dynamically adding Expectations to a suite but if an exception occurs while running the validation operator, all remaining expectations will not run.

Is there a way to validate configurations prior to validator.run_validation_operator()? Alternatively, is it possible to use the V2 Batch Kwargs in the meantime; could I achieve the same thing?

Thanks again.

**Comment by talagluck** [comment]
Hi @KentonParton - thanks for your response. How are you dynamically adding Expectations to a suite? If you add Expectations in interactive mode (against a batch of data, like with our CLI notebook workflow, e.g. `validator.expect_column_values_to_not_be_null`), then you should see results immediately. It's a rough work-around, but you could wrap statements like this into a `try`/`except` block which may solve your issue.

I do believe that this should work with the V2 Batch Kwargs as well, so that would be another possibility here. Please let me know if you have any trouble setting either of these up.

**Comment by KentonParton** [comment]
@talagluck this is what I have as a work around.

```python
    def process(self):
        data_context_config = self.set_data_context_config()
        context = BaseDataContext(project_config=data_context_config)

        batch_df = self.load_batch()
        runtime_batch_request = self.create_runtime_batch_request(batch_df)
        results = []
        failure_results = []
        for expectation in self.rules_list:
            try:
                suite: ExpectationSuite = context.create_expectation_suite("suite", overwrite_existing=True)
                expectation_configuration = ExpectationConfiguration(
                    expectation_type=expectation["method_name"],
                    kwargs=expectation["kwargs"],
                    meta=expectation["meta"]
                )
                suite.add_expectation(expectation_configuration=expectation_configuration)

                validator = context.get_validator(
                    batch_request=runtime_batch_request,
                    expectation_suite=suite,
                ).validate()
                results.extend(self.format_results(validator.results))
            except Exception as ex:
                failure_results.append(self.capture_exception_details(expectation["meta"], ex.message, traceback.format_exc()))
                continue

        return results, failure_results
```

If you could keep me updated on this issues expected completion date that would be great. It will allow my team and I to plan accordingly. Thanks for your help.

**Comment by AntonBiryukovUofC** [comment]
@talagluck I can confirm that if catch_exceptions is persisted as a config parameter in the serialized exception suite (the json file), then upon passing its name into a checkpoint, the Exception is raised and the validation essentially halts.

Our use case is similar to @KentonParton except that we use v3 objects (and validate via checkpoint runs).

**Comment by AntonBiryukovUofC** [comment]
A pretty nasty bug that defeats the purpose of having both table and column level expectations within the same suite...

**Comment by talagluck** [comment]
@KentonParton - thanks so much for sharing the follow-up. We will absolutely be in touch when we have an expected completion date for this.

@AntonBiryukovUofC - thank you for the follow-up and corroboration. Are you saying that not only does `catch_exceptions` not work, but if present, it will cause Checkpoints to fail? Would please describe this in a bit more detail? Also, how does this relate to having table and column level expectations within the same suite?

**Comment by jacylnkan** [comment]
@talagluck I think I can provide some clarification, as I am working with @AntonBiryukovUofC on the same project. `catch_exceptions` does NOT cause the Checkpoint to fail on its own, it just does not work in actually preventing the exception from killing the rest of the validation process. 

You can imagine this is a problem if we have table and column-level expectations within the same suite because say, we have an expectation that our columns match some given set (table-level). However, later on in the same suite, we have our individual column expectations. Our column expectations will never run if the table-level raises an exception. So in the end, we will have to ensure first that our table-level expectations pass (by putting it in one suite), then running the column-level expectations after (putting them in another suite).

**Comment by talagluck** [comment]
Hi @jacylnkan - thanks for the follow-up. It is not uncommon to split up expectations into multiple suites for different reasons, although I agree this is not an ideal reason. To confirm, are your table-level expectations more likely to fail than your column-level expectations? I'm interested in better understanding that distinction.

**Comment by jacylnkan** [comment]
@talagluck The issue is that an exception is raised if (for example) I am missing a column from my data that I had written an expectation for. Ideally, I would like for it to fail the expectation and move on to other expectations. However, it raises an exception and I cannot use `catch_exceptions` to prevent it from killing the rest of the validation process. It's not that the table-level expectation are _more likely_ to fail, but that the failure of the table-level expectation prevents any other expectations from running at all.

**Comment by talagluck** [comment]
I see - thank you for the clarification, @jacylnkan! I am hoping to have a better estimate for the resolution of this issue next week.

**Comment by KentonParton** [comment]
@talagluck has there been any progress on this?

**Comment by talagluck** [comment]
Hi @KentonParton - thanks for checking in, and I'm sorry that I haven't updated here. This has been top of mind for us, but it hasn't yet reached a solution. A preliminary spike found that this issue went deeper, and so would take more time to address, than expected. Tomorrow we'll be discussing the best way to move forward here, and exactly how to prioritize this, and I'll make sure to update you then.

**Comment by talagluck** [comment]
Hi @KentonParton - thank you again for your patience here. I've touched base with the team, and this issue is currently prioritized. It's more complex than expected, and so may take a few weeks, but we have engineers working full-time on this until it is completed. We expect that it will be finished by the end of our next sprint, at the latest.

I'll be OOO for a couple of weeks beginning tomorrow, and so I'm tagging @alexsherstinsky here, who will be the primary engineer working on this, in case anything else comes up.

Again, apologies for the delays, and we appreciate you raising the issue, as well as your patience!

**Comment by KentonParton** [comment]
Thanks for the update @talagluck

**Comment by alexsherstinsky** [comment]
Hello @KentonParton and @talagluck -- this issue has been fixed -- the improved behavior will be part of the next release.  Thank you.

**Comment by KentonParton** [comment]
That's great news @alexsherstinsky. Thank you!

**Comment by NathanFarmer** [comment]
I'm closing this out, as it was addressed by PR #3433.
